src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,120 @@
+import { Comment } from '../types/Comment';
+import classNames from 'classnames';
+
+type Props = {
+  postId: number;
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ postId, setComments }) => {
+  const [currentName, setCurrentName] = useState('');
+  const [currentEmail, setCurrentEmail] = useState('');
+  const [currentBody, setCurrentBody] = useState('');
+  const [isAddingComment, setIsAddingComment] = useState(false);
+  const [isNameError, setIsNameError] = useState(false);
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [isBodyError, setIsBodyError] = useState(false);
+
+  const resetForm = (reason: 'submit' | 'clear') => {
+    if (reason === 'clear') {
+      setCurrentName('');
+      setCurrentEmail('');
+    }
+
+    setCurrentBody('');
+    setIsNameError(false);
+    setIsEmailError(false);
+    setIsBodyError(false);
+  };
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCurrentName(event.target.value);
+    if (isNameError) {
+      setIsNameError(false);
+    }
+  };
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCurrentEmail(event.target.value);
+    if (isEmailError) {
+      setIsEmailError(false);
+    }
+  };
+
+  const handleBodyChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setCurrentBody(event.target.value);
+    if (isBodyError) {
+      setIsBodyError(false);
+    }
+  };
+
+  const validateName = (testedName: string) => {
+    if (testedName.trim() === '') {
+      setIsNameError(true);
+    }
+
+    return testedName.trim().length > 0;
+  };
+
+  const validateEmail = (testedEmail: string) => {
+    if (!/\S+@\S+\.\S+/.test(testedEmail)) {
+      setIsEmailError(true);
+    }
+
+    return /\S+@\S+\.\S+/.test(testedEmail);
+  };
+
+  const validateCommentBody = (testedBody: string) => {
+    if (testedBody.trim() === '') {
+      setIsBodyError(true);
+    }
+
+    return testedBody.trim().length > 0;
+  };
+
+  const validateForm = () => {
+    validateName(currentName);
+    validateEmail(currentEmail);
+    validateCommentBody(currentBody);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    validateForm();
+
+    if (
+      validateName(currentName) &&
+      validateEmail(currentEmail) &&
+      validateCommentBody(currentBody)
+    ) {
+      setIsAddingComment(true);
+    }
+  };
+
+  useEffect(() => {
+    if (!isAddingComment || isNameError || isEmailError || isBodyError) {
+      return;
+    }
+
+    newComment({
+      name: currentName,
+      email: currentEmail,
+      body: currentBody,
+      postId: postId,
+    }).then((data: unknown) => {
+      const dataAsComment = data as Comment;
+
+      setComments(prevComments => [...prevComments, dataAsComment]);
+      resetForm('submit');
+      setIsAddingComment(false);
+    });
```

Comment:
The `newComment` API call should handle potential errors. Consider adding a `.catch` block to handle any errors that might occur during the API call and update the UI accordingly.